core/chains/evm/config/toml/config.go

```diff
@@ -432,6 +434,57 @@ func (t *Transactions) setFrom(f *Transactions) {
 	if v := f.ResendAfterThreshold; v != nil {
 		t.ResendAfterThreshold = v
 	}
+	t.AutoPurge.setFrom(&f.AutoPurge)
+}
+
+type AutoPurgeConfig struct {
+	AutoPurgeStuckTxs        *bool
+	AutoPurgeThreshold       *uint32
+	AutoPurgeMinAttempts     *uint32
+	AutoPurgeDetectionApiUrl *commonconfig.URL
+}
+
+func (a *AutoPurgeConfig) ValidateConfig() (err error) {
+	if a.AutoPurgeStuckTxs == nil || !*a.AutoPurgeStuckTxs {
+		return
+	}
+
+	if a.AutoPurgeDetectionApiUrl != nil && !a.AutoPurgeDetectionApiUrl.IsZero() {
+		switch a.AutoPurgeDetectionApiUrl.Scheme {
+		case "http", "https":
+			return
+		default:
+			return commonconfig.ErrInvalid{Name: "AutoPurgeDetectionApiUrl", Value: a.AutoPurgeDetectionApiUrl.Scheme, Msg: "must be http or https"}
+		}
+	}
+
+	if a.AutoPurgeThreshold == nil {
```

Comment:
nit: optional, but if you collapse the null and 0 checks in same line, then code is less, and more readable

Reply:
I kept them separate so I could differentiate between `ErrMissing` and `ErrInvalid` but I can maybe combine these into just `ErrInvalid`